devito/function.py

```diff
@@ -438,6 +438,7 @@ def __init__(self, *args, **kwargs):
             self._padding = padding
 
             # Dynamically add derivative short-cuts
+            self.derivatives = self.derivatives if hasattr(self, 'derivatives') else []
```

Comment:
I really don't like this. The presence of ``hasattr`` is a symptom we're doing this in the wrong place (do we need an extra class?). Why was this necessary?

Reply:
btw, why do we even need this instance attribute ``derivatives``? when do we need it?

Reply:
So that people can access easily the list of available shortcuts.